### ISSUE TITLE
No page can load while uBlock is active

### ISSUE THREAD

**Issue by j36e** [uBlockOrigin/uBlock-issues#3217]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

https://github.com/uBlockOrigin/uBlock-issues/assets/147976004/5633061e-c972-4d8e-87d6-497ac5664927

Pages won't load with UBO enabled but everything works fine when it's off. Everything was fine a couple of hours ago and yes I did reboot my computer.
Sorry for the bad video quality my laptop is slow

### A specific URL where the issue occurs.

```yaml
https://github.com/uBlockOrigin/uBlock-issues/issues/new?assignees=&labels=&projects=&template=bug_report.yml
```

### Steps to Reproduce

1. [Enable UBO]
2. [Try to load a page]

### Expected behavior

The page to load

### Actual behavior

It did not :(

### uBO version

1.57.0

### Browser name and version

Chromium (chromium-123.0.6312.86_1)

### Operating System and version

Void linux

**Comment by j36e** [comment]
The first tab title seemed to load but it was probably cached since I tried the same query minutes before

**Comment by j36e** [comment]
Lmao the addon wouldnt do anything when I clicked on it so I reinstalled and it's all working now sorry for the stupid issue report

**Comment by gorhill** [comment]
We had other reports of such behavior, so I don't consider this invalid. Problem is that so far I have been unable to investigate further since the issue disappear upon reinstall.

@j36e Do you think that uBO had just updated to 1.57.0 before the issue occurred?

**Comment by stephenhawk8054** [comment]
@voidlyvidevoid Can you do the steps described in this: https://github.com/uBlockOrigin/uBlock-issues/issues/3217#issuecomment-2063911578

**Comment by garry-ut99** [comment]
> j36e : No page can load while uBlock is active
> j36e : Pages won't load with UBO enabled 
> j36e : Possible in https://github.com/uBlockOrigin/uBlock-issues/issues/3219, OP did face something similar (chromium used)

More likely in this one: 

> 
> - https://github.com/uBlockOrigin/uBlock-issues/issues/3198 : (spain): "y no me carga ninguna web se queda cargando" :
> (enlgish): "and it doesn't load any web site it just keeps loading"

<hr>

> voidlyvidevoid : `allReadyAfter` after this incident was over 80k ms

In this one 85k ms too: 
- https://github.com/uBlockOrigin/uAssets/issues/23344

also you can read (if you didn't see them before):
- https://github.com/uBlockOrigin/uAssets/discussions/16939
- https://github.com/uBlockOrigin/uBlock-issues/issues/1865

**Comment by stephenhawk8054** [comment]
@voidlyvidevoid Can you click on uBO icon > &#128172; [Report button](https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface#the-tools) > scroll to the bottom > click on `Troubleshooting Information` and copy paste it here?

**Comment by stephenhawk8054** [comment]
This gitlab user is removed by Gitlab already so its filter list cannot be updated: `https://gitlab.com/magnolia1234/bypass-paywalls-clean-filters/-/raw/main/bpc-paywall-filter.txt`

You are using a heavily modified configuration so we cannot pinpoint what causes the issue. Backup and reset to default settings and test again. Then restore your configuration one-by-one and see which one is causing issue.

Anyways, since you can access uBO's dashboard, I don't think your issue is the same as this issue. It's more similar to this: https://github.com/uBlockOrigin/uBlock-issues/issues/1865 so please continue in that thread instead. But do what I said above first.

**Comment by stephenhawk8054** [comment]
However, I'm thinking maybe some user-removed gitlab links could be related to OP's issue? From the screenshot, it shows `gitlab.com/users/sign_in` link in the error line instead.

**Comment by gorhill** [comment]
I expect over half a million network filter is going to cause longer delay at launch, although 25s seems high, it's not a configuration I test often. It could be a storage issue, in which case it might be useful to force the indexedDB to be re-created, a backup/restore should do this.

**Comment by gorhill** [comment]
> there is a timeout for the badge

The countdown starts when uBO is all ready if I remember correctly.

**Comment by garry-ut99** [comment]
> there is a timeout for the badge - it will be shorter than his 80 seconds.

I'm not talking about a badge, but about an icon, they are two separate things, the yellowish icon has no timeout, and as for the the yellowish badge, like gorhill mentioned, its countdown starts _after_  uBO is ready, like already mentioned 2 times in Wiki:

> https://github.com/gorhill/uBlock/wiki/Toolbar-icon : The yellowish badge `!` will persist **once uBO is ready** to filter properly (...)
> https://github.com/gorhill/uBlock/wiki/Toolbar-icon#important-note : The sticky yellowish badge **once uBO is fully loaded**

also worth to mention another difference between the yellowish icon and the yellowish badge: the yellowish badge doesn't prevent uBO and dashboard from working, which means uBO and dashboard are accessible after yellowish icon disappears, even if yellowish badge is still here, because the yellowish badge servers only as an additional non-blocking warning.

**Comment by Avellea** [comment]
Popping in to say I have the same issue on the latest versions of Chrome and Thorium. I cannot access the uBO dashboard or any websites. There is nothing reported in the uBO DevTools console either.

**Comment by stephenhawk8054** [comment]
@Avellea Do you remember you were using / enabling any additional filter lists (bypass paywall...)?

**Comment by Avellea** [comment]
> @Avellea Do you remember you were using / enabling any additional filter lists (bypass paywall...)?

I do not use any custom filter lists at all.

**Comment by garry-ut99** [comment]
"Popping in" is appreciated and useful to know how many users have the same problem, and how big is the problem, but we need more details to help:

>  I cannot access the uBO dashboard or any websites. 

When has it happened exactly?: 
- after uBO re/installation
- after browser re/start
- after reenabling uBO
- while browsing internet as usual

Has been uBO's icon yellowish while as per https://github.com/uBlockOrigin/uBlock-issues/wiki/Toolbar-icon while you were unable to access dashboard?

> Uninstalling and reinstalling the extension does seem to fix the issue.

Are you able to reproduce the issue again or it doesn't occur anymore.
If you could reproduce it again it would be helpful for us to diagnose it.

What is your OS and version of uBO.

**Comment by Avellea** [comment]
> "Popping in" is appreciated and useful to know how many users have the same problem, and how big is the problem, but we need more details to help:

Apologies, let me clear this up.

- Issue occurred after a browser restart.
- The toolbar icon was the regular red icon
- I have not been able to reproduce the issue since the fix.
- Version was the latest at the time, uBlock Origin 1.57.0

**Comment by iMarbot** [comment]
Started experiencing this yesterday with uBlock Origin development build `1.57.3.6`. I'm sorry but I don't know whether it happened after a browser or extension update. It happened after I closed Chrome entirely and then re-opened it, everything was refusing to load. The icon was yellow with the exclamation mark. Also, the extension was using almost 100 CPU in Chrome's task manager.

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/5881994/647735f7-cd8c-4924-a8f1-6336cd59e483)

Disabling the extension leads to pages instantly loading, without even needing to refresh or restart their loading.

Chrome version: `124.0.6367.91 (Official Build) (64-bit) (cohort: M124 Rollout)`

I have not tested cleaning the extension data or re-installing it as I didn't want to lose my settings but also in case I can help reproduce the issue.

I also have uBlock Origin `1.57.0` installed (not running at the same time, obviously) which works fine on the exact same browser environment. The settings are a little different but I can't even open the uBlock popup or settings in the dev build because it's just hanging on something with that high CPU usage.

Let me know if there's anything I have missed or can do to help isolate the issue.

**Comment by stephenhawk8054** [comment]
@iMarbot Can you do this step: https://github.com/uBlockOrigin/uBlock-issues/issues/3217#issuecomment-2063911578 ?

**Comment by iMarbot** [comment]
> @iMarbot Can you do this step: [#3217 (comment)](https://github.com/uBlockOrigin/uBlock-issues/issues/3217#issuecomment-2063911578) ?

Nope, it's empty. However, I'm not sure if it outputs something before I'm able to open the window with the console.

**Comment by stephenhawk8054** [comment]
@iMarbot Which OS are you using?

Can you go to `chrome://flags/#enable-tls13-kyber`, disable `TLS 1.3 hybridized Kyber support`, restart the browser (make sure that all instances of chrome is turned off in Task Manager) and test again if the issue still occurs?

**Comment by WesselKroos** [comment]
I had this yesterday with default settings.
Eventually got it fixed after doing these steps:
- disable the extension
- re-enabling the extension
- go straight to the options (dashboard.html) 
- force an update on every enabled filter list
- Wait for all filters to be updated
- (optional) disable and re-enable the extension again

I guess a filter list was causing this problem for me or something got corrupted in a filter list? I don't know which one specifically.

**Comment by garry-ut99** [comment]
> iMarbot : However, I'm not sure if it outputs something before I'm able to open the window with the console.

It will, but only in _uBO dev_, because debugging information to console output is enabled by default only in _uBO dev_.

> iMarbot : Nope, it's empty. 

In which version of your uBO it is empty: `1.57.3.6` (dev channel) or on  `1.57.0` (stable channel)?

If it's empty for you on _uBO dev_, then this is definately abnormal behaviour - it would indicate that the issue happens at very early stage of uBO initialization, as soon as uBO is enabled, preventing uBO even from debugging information to console output.

**Comment by iMarbot** [comment]
> Can you go to `chrome://flags/#enable-tls13-kyber`, disable `TLS 1.3 hybridized Kyber support`, restart the browser (make sure that all instances of chrome is turned off in Task Manager) and test again if the issue still occurs?

```
Edition		Windows 11 Home
Version		23H2
Installed on	‎16/‎06/‎2023
OS build	22635.3566
Experience	Windows Feature Experience Pack 1000.22700.1007.0
```

I was going to report back with good news initially, disabling that flag and relaunching chrome fixed the problem. However, after resetting it back to default and then launching the browser again, the issue did not return. I'm not sure if that is expected or not. And to top it off, when I relaunched chrome, it updated to version `124.0.6367.119 (Official Build) (64-bit) (cohort: Stable)`, so potentially that might've fixed it instead.

----

> In which version of your uBO it is empty: 1.57.3.6 (dev channel) or on 1.57.0 (stable channel)

I checked both and both were empty, but obviously I reported back about the dev build as that's the one that was experiencing the issue. After the issue went after with one of the above reasons (or something else, I guess?), the console has a bunch of debugging information in it.

**Comment by stephenhawk8054** [comment]
@iMarbot So currently does the issue occur any more?

**Comment by Maxr1998** [comment]
Had this happen to me on Vivaldi (6.7.3329.24) on Linux (Arch) this morning.

<details>
<summary>Troubleshooting information</summary>

uBlock Origin: 1.57.0
Chromium: 124
filterset (summary):
 network: 145821
 cosmetic: 161447
 scriptlet: 33960
 html: 0
listset (total-discarded, last-updated):
 added:
  https://filters.adtidy.org/extension/ublock/filters/14.txt: 78408-4449, 4d.9h.30m
  fanboy-cookiemonster: 50983-204, 1d.11h.4m
  fanboy-thirdparty_social: 68-0, 1d.23h.5m
  ublock-annoyances: 6966-28, 2m Δ
 default:
  user-filters: 15-0, never
  DEU-0: 7439-1, 12h.15m
  easylist: 86828-469, 2m Δ
  easyprivacy: 50856-799, 2m Δ
  plowe-0: 3735-0, 3d.10h.44m
  ublock-badware: 8505-0, 2m Δ
  ublock-filters: 37812-426, 2m Δ
  ublock-privacy: 881-2, 2m Δ
  ublock-quick-fixes: 261-5, 2m Δ
  ublock-unbreak: 2268-21, 2m Δ
  urlhaus-1: 12765-0, 12h.15m
filterset (user): [array of 17 redacted]
trustedset:
 added: [array of 36 redacted]
switchRuleset:
 added: [array of 2 redacted]
hostRuleset:
 added: [array of 1 redacted]
userSettings:
 advancedUserEnabled: true
 cloudStorageEnabled: true
hiddenSettings: [none]
supportStats:
 allReadyAfter: 757 ms
 maxAssetCacheWait: 208 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 21
 network:
  github.com: 21
</details>

No pages would load, and CPU usage was very high. Restarting the browser multiple times didn't help. Found uBlock at 100% in the task manager (thanks to a hint in [this reddit thread](https://www.reddit.com/r/vivaldibrowser/comments/1cgf1ub/vivaldi_not_loading_pages/)), after disabling the extension, CPU usage went down and pages would load again.

Re-enabling the extension caused the issue to return. I cleared the browser cache and restarted the browser once more, re-enabled the extension, but the issue persisted. Disabled and re-enabled the extension *again*, and it suddenly started to work. Unfortunately, I don't see anything in the `background.html` console.

**Comment by garry-ut99** [comment]
> Unfortunately, I don't see anything in the `background.html` console.

Because as already explained 2 comments above ( in https://github.com/uBlockOrigin/uBlock-issues/issues/3217#issuecomment-2088560489 ) : the stable version of your uBO (1.57.0) doesn't output anything to the console by default, until you manually set a flag.

Besides, a new verison of uBO has just been released: `1.57.3b7` which contains a fix: https://github.com/gorhill/uBlock/commit/c098eb8625.

**Comment by gorhill** [comment]
Regarding the above commit: So far I have been unable to observe the [telltale code path being triggered after days](https://github.com/gorhill/uBlock/blob/1.57.2/src/js/s14e-serializer.js#L600) (by setting a breakpoint) -- though the browser was not in heavy use, that code path I gather could have been visited when a selfie is generated, but no luck so far. To increase likelihood I have enabled many other stock lists.

Whoever with JS dev skill is suffering an issue of broken uBO, a breakpoint at <https://github.com/gorhill/uBlock/blob/1.57.2/src/js/s14e-serializer.js#L744-L745> being triggered at launch time could confirm the issue is related to the above commit.

**Comment by Ovear** [comment]
I encountered the same problem on Arch Linux after upgrading the system today.

uBlock will consume 100% cpu once trying to load any page (and prevent page from loading), same situation happened even in incognito mode. 

Turn off uBlock makes page to load immediately and CPU dropped to normal level. Not observed any output in Console of uBlock's background page.

Before upgrading Chrome, this situation happened randomly with some pages stop loading but will work with a force close and open again. Not sure if it's related to this problem.

```
upgraded google-chrome (123.0.6312.122-1 -> 124.0.6367.155-1)
```

The solution to this (currently) also weird, I disabled `Enable my custom filters`, and everything back to work, even I enable it again after that.

**Comment by Maxr1998** [comment]
> Because as already explained 2 comments above ( in https://github.com/uBlockOrigin/uBlock-issues/issues/3217#issuecomment-2088560489 ) : the stable version of your uBO (1.57.0) doesn't output anything to the console by default, until you manually set a flag, which I suppose you didn't.

Nope, I must have missed that comment and didn't manually set the flag. I haven't been able to reproduce the issue anymore since then, but I'll remember it if I notice it again.

**Comment by Procyon-b** [comment]
Just to let everyone here know that I have an installation that is blocked like some users are reporting.

Chrome 109 (windows 8.1, so no choice) If it can be useful to use it for tests feel free to ask. (I won't be 7/7 24/24 behind the screen)

I'm currently using this profile with uBO 1.57.3.8 disabled, and with beta 1.57.6 side-loaded.

The symptoms were/are:
no access to any site (time-out) even on my local network. I though it was the wifi router. Tried with FF. It worked partially (could not access wifi router). Restarted both browsers. Same issues. Rebooted. "Looked" better but uBO's icon stayed yellow with "!". High CPU (didn't investigate further). Disable/enable uBO. Icon is red. Able to open the popup, but as soon as I try to access a site, uBO is unresponsive. Disable/enable. Open uBO dashboard. Export settings (already had backups).

Tried again right now (after uBO sideloaded) : 1.57.3.8 is still in distress.

I have backuped the chrome user dir. So hopefully I can recreate the problem if one of my actions solves the issue.

**Comment by iMarbot** [comment]
@Procyon-b maybe test the stuff that was suggested to me (checking for console output in `background.html` (if on dev channel) and setting `chrome://flags/#enable-tls13-kyber` to disabled.

**Comment by Procyon-b** [comment]
@iMarbot I don't really care about solving my issue. It's already solved.
It's just to inform that I have an installation that can be used to debug the root of the issue.

**Comment by iMarbot** [comment]
> @iMarbot I don't really care about solving my issue. It's already solved. It's just to inform that I have an installation that can be used to debug the root of the issue.

And testing both of those things and seeing whether they fix the issue or not (or have information that might point to what the issue is) should help narrow down the cause. But I'll leave that to the uBlock devs.

**Comment by gorhill** [comment]
> I have an installation that can be used to debug the root of the issue.

Is it something you would share (zip file) with me to see if I can reproduce on my side?

**Comment by Procyon-b** [comment]
@gorhill yes , f course. That's why I mentionned it.

**Comment by gorhill** [comment]
Thanks, you can send the file to rhill at raymondhill dot net.

**Comment by Procyon-b** [comment]
Is there a size limit by email ?
Right now my complete profile (without the various cache dirs) is 124Mb. I'll try removing unecessary things (other extensions) as long as uBO is still broken.
Do you need the non-profile files too (the ones not in a profile dir) ?

**Comment by gorhill** [comment]
I would just need what is related to uBO, nothing else. Maybe search for uBO's extension id (`cjpalhdlnbpafiamejdnhcphjbkeiagm`) and put all the matching folders in a zip file?

**Comment by Procyon-b** [comment]
no prob.

**Comment by Procyon-b** [comment]
It will be `cgbcahbpdhpcegmbfconppldiemgcoii` since the problem is with the beta version (just after updating x.6 to x.8)

**Comment by gorhill** [comment]
I received 3 emails, it looks they all contain the same attachments?

**Comment by Procyon-b** [comment]
It's a 3 parts rar attachment (my provider doesn't allow files bigger than 20Gb)
The 4th email is bouncing (no attach, text only)

**Comment by Procyon-b** [comment]
File names should be:
```
Default.part1.rar  18,000,000 bytes
Default.part2.rar  18,000,000 bytes
Default.part3.rar  12,210,005 bytes
```
rar.exe should recontruct while decompressing simply by giving it the 1st file

**Comment by gorhill** [comment]
I can reproduce 100 CPU% at launch if I open a webpage once uBO is ready. The issue seems to be that the selfie is "corrupted". I use quotes because it seems valid, but one part of the selfie (which is saved in multiple parts to indexedDB) seems to not correspond to the rest, as if an older version of that part was used along with the other proper up to date parts. I still need to investigate, and think of a way to detect such bad state.

**Comment by Procyon-b** [comment]
Good to know that it is useful.

I don't know if this can be useful. I forced the update today. Chrome was running for about a week now. At night, I put windows in deep sleep instead of closing all progs, and I regularly start an update of the extension which on chrome will force updates. That's what happened today.

I don't know if this can explain why the selfie can get in that weird state.
I don't remember having a crash recently, so...

**Comment by Procyon-b** [comment]
I have updated the extension sideload to `.8` and I see this on the background devtools console (opened by curiosity):

![uBO-1 57 3 8-bgerrors](https://github.com/uBlockOrigin/uBlock-issues/assets/40043075/67b2ea78-1959-4738-95cc-9b008313f011)

note: I haven't checked in `.6` if it made errors too.

**Comment by gorhill** [comment]
What version of Chrome/Chromium is this? Now I see that [`alarms.create`](https://developer.chrome.com/docs/extensions/reference/api/alarms#method-create) accepts a third parameter only starting with Chrome 111+. Any earlier version of Chrome would suffer the issue you report -- possibly explaining people unable to run uBO in some oldish Chromium versions.

**Comment by Procyon-b** [comment]
version 109 on windows 8.1. Can't go higher.

**Comment by garry-ut99** [comment]
> version 109 on windows 8.1. Can't go higher.

There are unofficial builds/forks of Chromium, Edge, Brave, Firefox, Thunderbird, Electron and so on, which can run on Win 7 / 8 (and sometimes even on XP): [Supremium 122](https://github.com/win32ss/supermium/releases), [R3dfox 125.0.3](https://github.com/Eclipse-Community/r3dfox/releases), [Firefox / Brave / Librewolf / Thunderbird / Electron](https://github.com/adeii/supermium-portable/releases/).  I use them, like mentioned in https://github.com/uBlockOrigin/uBlock-issues/issues/3138#issuecomment-2016803951.

**Comment by heathlarsen** [comment]
Hey, just I'm also experiencing this issue and thought I would add my 2c.

I'm using Firefox and have the latest uBO pre-release [1.57.3b10](https://github.com/gorhill/uBlock/releases/tag/1.57.3b10). I only have uBO and BitWarden password manager extensions installed. This is happening only in incognito mode.

I am able to reproduce this issue pretty reliably. Within an incognito tab, opening my BitWarden extension by clicking it's icon in the pinned extensions. I immediately notice the extension doesn't load, and then now browser can no longer load pages and Firefox's native Task Manager reports 'Extensions' are using 100% CPU.

**Comment by gorhill** [comment]
> Firefox's native Task Manager reports 'Extensions' are using 100% CPU.

You need to use the [profiler](https://profiler.firefox.com/) and report results to find out whether this is caused by uBO (and if so, which code in uBO) or something else since all extensions share the same process in Firefox.

**Comment by heathlarsen** [comment]
> You need to use the [profiler](https://profiler.firefox.com/) and report results to find out whether this is caused by uBO (and if so, which code in uBO) or something else since all extensions share the same process in Firefox.

Yeah I've tried this. Unfortunately when this bug is triggered even the profiler results page won't load. :(

**Comment by gorhill** [comment]
This happens only if BitWarden is active and you click its toolbar icon?

**Comment by heathlarsen** [comment]
> This happens only if BitWarden is active and you click its toolbar icon?

I'm not sure if thats the _only_ way I would be able to trigger it, but it's the only way I've found so far. Yes, these are the only two extensions I have enabled and I'm using the BitWarden icon in the toolbar (to the right of the address bar).

> Might be related to [bitwarden/clients#8729](https://github.com/bitwarden/clients/issues/8729). Notice that this happens with other content blockers too, not just uBO.

Yes this looks like similar to what I'm experiencing.

- I've found that it requires me to be in incognito mode for about ~1 minute. Prior to that ~1 minute time elapsing I can open BitWarden as many times as I can and navigate around anywhere freely.
- If I don't open BitWarden then I can continue to browse freely beyond this time
- As soon as I open BitWarden after that ~1 min I will get that grey (sometimes white) blank box from the BitWarden issue's screenshot. From that point on no pages will load and the only way to fix this is to restart Firefox entirely (Incognito and Normal windows included).
- If I do NOT have uBO running then BitWarden will still break after ~1 min but my browsing is unaffected, only BitWarden is broken.

**Comment by heathlarsen** [comment]
> This happens only if BitWarden is active and you click its toolbar icon?

I'm not sure if thats the _only_ way I would be able to trigger it, but it's the only way I've found so far. Yes, these are the only two extensions I have enabled and I'm using the BitWarden icon in the toolbar (to the right of the address bar).

> Might be related to [bitwarden/clients#8729](https://github.com/bitwarden/clients/issues/8729). Notice that this happens with other content blockers too, not just uBO.

Yes this looks like similar to what I'm experiencing.

---

I've done some more testing:
- For this bug to occur it requires me to be in incognito mode for about ~1 minute. Prior to that ~1 minute time elapsing I can open BitWarden as many times as I can and navigate around anywhere freely.
- If I don't open BitWarden then I can continue to browse freely beyond this time
- As soon as I open BitWarden after that ~1 min I will get that grey (sometimes white) blank box from the BitWarden issue's screenshot. From that point on no pages will load and the only way to fix this is to restart Firefox entirely (Incognito and Normal windows included).
- If I do NOT have uBO running then BitWarden will still break after ~1 min but my browsing is unaffected, only BitWarden is broken.

**Comment by gorhill** [comment]
> only BitWarden is broken

A broken BitWarden might be a sign that it's hogging the extensions process, which would prevent uBO from doing it's work, which would cause webpages from loading since it's waiting on uBO. See if you have the same issue with other content blockers, and if so, the issue is not uBO-specific.

**Comment by heathlarsen** [comment]
You seem to be correct, it's entirely BitWarden's fault.
- After testing with with ABP I saw the exact same behaviour, pages did not load.
- With no ad blocker, once BitWarden breaks the 'Extension' CPU usage is maxed.

It'll be interesting to see what other people in this issue thread say when they get a chance to reply.

**Comment by Procyon-b** [comment]
@gorhill
I have updated right now after seeing your comment. A couple minutes after the update I see this in b10's console (saved console log to file and pasted content below). I saw 5 errors. Note that the order seem different here compared to the devtools window.
```
console.js:38 [uBO] Start sequence of loading storage-based data 135 ms after launch
console.js:38 [uBO] Admin settings ready 243 ms after launch
console.js:38 [uBO] Hidden settings ready 255 ms after launch
console.js:38 [uBO] Extra admin settings ready 258 ms after launch
console.js:38 [uBO] Backend storage for cache will be indexedDB
console.js:38 [uBO] Version ready 263 ms after launch
console.js:38 [uBO] cacheStorage.get: idbStorage assetCacheRegistry
console.js:38 [uBO] First fetch ready 270 ms after launch
console.js:38 [uBO] cacheStorage.get: idbStorage cache/selfie/public_suffix_list.dat
console.js:38 [uBO] cacheStorage.get: idbStorage compiledMagic,selfieMagic
console.js:38 [uBO] List selection ready 280 ms after launch
console.js:38 [uBO] User settings ready 282 ms after launch
console.js:38 [uBO] Suspend network activity listener at 284 ms
console.js:38 [uBO] Loaded assetCacheRegistry
console.js:38 [uBO] cacheStorage.set: assetCacheRegistry
console.js:38 [uBO] cacheStorage.get: storage.local compiledMagic,selfieMagic
console.js:38 [uBO] Cache magic numbers ready 427 ms after launch
console.js:38 [uBO] Loaded PSL from selfie
console.js:38 [uBO] PSL ready 473 ms after launch
console.js:38 [uBO] cacheStorage.get: idbStorage cache/selfie/staticMain
console.js:38 [uBO] cacheStorage.get: idbStorage cache/selfie/staticExtFilteringEngine
console.js:38 [uBO] cacheStorage.get: idbStorage cache/selfie/staticNetFilteringEngine
console.js:38 [uBO] cacheStorage.get: idbStorage cache/selfie/redirectEngine/resources
console.js:38 [uBO] cacheStorage.get: storage.local cache/selfie/redirectEngine/resources
console.js:38 [uBO] cacheStorage.set: cache/selfie/redirectEngine/resources
console.js:38 [uBO] Loaded filtering engine from selfie 1192 ms after launch
console.js:38 [uBO] All ready 1230 ms (selfie) after launch
console.js:38 [uBO] cacheStorage.get: idbStorage assetSourceRegistry
console.js:38 [uBO] Loaded assetSourceRegistry
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at Object.create (vapi-background.js:1787:23)
    at µb.scheduleAssetUpdater (storage.js:1586:21)
    at start.js:534:8
(anonymous) @ webext.js:35
console.js:38 [uBO] cacheStorage.set: assetCacheRegistry
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at vapi-background.js:1792:27
(anonymous) @ webext.js:35
console.js:38 [uBO] Updater: cycle start
console.js:38 [uBO] Updater: prefer CDNs
console.js:38 [uBO] Diff updater: cycle start
console.js:38 [uBO] Diff updater: hard updating NLD-0
console.js:38 [uBO] Diff updater: NLD-0 / patches/2024.1.16.800.patch#easydutch / nopatch
console.js:38 [uBO] Diff updater: soft updating ublock-quick-fixes,ublock-badware,easyprivacy,easylist,ublock-privacy,ublock-annoyances,ublock-unbreak,ublock-filters
console.js:38 [uBO] Diff updater: skip update of ublock-quick-fixes using ../patches/2024.5.13.1281.patch#ublock-quick-fixes
	reason: nodiff
console.js:38 [uBO] Diff updater: skip update of ublock-badware using ../patches/2024.5.13.1281.patch#ublock-badware
	reason: nodiff
console.js:38 [uBO] Diff updater: skip update of easyprivacy using ../patches/2024.5.13.1281.patch#easyprivacy
	reason: nodiff
console.js:38 [uBO] Diff updater: skip update of easylist using ../patches/2024.5.13.1281.patch#easylist
	reason: nodiff
console.js:38 [uBO] Diff updater: skip update of ublock-privacy using ../patches/2024.5.13.1281.patch#ublock-privacy
	reason: nodiff
console.js:38 [uBO] Diff updater: skip update of ublock-annoyances using ../patches/2024.5.13.1281.patch#ublock-annoyances
	reason: nodiff
console.js:38 [uBO] Diff updater: skip update of ublock-unbreak using ../patches/2024.5.13.1281.patch#ublock-unbreak
	reason: nodiff
console.js:38 [uBO] Diff updater: skip update of ublock-filters using ../patches/2024.5.13.1281.patch#ublock-filters
	reason: nodiff
console.js:38 [uBO] Diff updater: cycle complete
console.js:38 [uBO] Updater: cycle end
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at Object.create (vapi-background.js:1787:23)
    at µb.scheduleAssetUpdater (storage.js:1586:21)
    at µb.assetObserver (storage.js:1676:14)
    at fireNotification (assets.js:238:19)
    at updateDone (assets.js:1440:5)
    at updateNext (assets.js:1394:16)
(anonymous) @ webext.js:35
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at vapi-background.js:1792:27
(anonymous) @ webext.js:35
(anonymous) @ webext.js:28
(anonymous) @ vapi-background.js:1792
Promise.then (async)
createIfNotPresent @ vapi-background.js:1790
(anonymous) @ storage.js:140
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
µb.incrementRequestStats @ storage.js:148
journalProcess @ pagestore.js:743
(anonymous) @ pagestore.js:352
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
offon @ vapi-common.js:64
journalAddRootFrame @ pagestore.js:685
onNavigation @ tab.js:929
onCommittedHandler @ vapi-background.js:614
onCommittedHandler @ vapi-background-ext.js:59
(anonymous) @ vapi-background.js:295
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at vapi-background.js:1792:27
(anonymous) @ webext.js:35
(anonymous) @ webext.js:28
(anonymous) @ vapi-background.js:1792
Promise.then (async)
createIfNotPresent @ vapi-background.js:1790
(anonymous) @ storage.js:140
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
µb.incrementRequestStats @ storage.js:148
journalProcess @ pagestore.js:743
(anonymous) @ pagestore.js:352
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
journalAddRequest @ pagestore.js:662
onBeforeRequest @ traffic.js:103
onBeforeSuspendableRequest @ vapi-background.js:1287
browser.webRequest.onBeforeRequest.addListener.denormalizeFilters.urls @ vapi-background.js:1239
subEventCallback @ extensions::webRequestEvent:106
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at vapi-background.js:1792:27
(anonymous) @ webext.js:35
(anonymous) @ webext.js:28
(anonymous) @ vapi-background.js:1792
Promise.then (async)
createIfNotPresent @ vapi-background.js:1790
(anonymous) @ storage.js:140
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
µb.incrementRequestStats @ storage.js:148
journalProcess @ pagestore.js:743
(anonymous) @ pagestore.js:352
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
journalAddRequest @ pagestore.js:662
onBeforeRequest @ traffic.js:103
onBeforeSuspendableRequest @ vapi-background.js:1287
browser.webRequest.onBeforeRequest.addListener.denormalizeFilters.urls @ vapi-background.js:1239
subEventCallback @ extensions::webRequestEvent:106
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at vapi-background.js:1792:27
(anonymous) @ webext.js:35
(anonymous) @ webext.js:28
(anonymous) @ vapi-background.js:1792
Promise.then (async)
createIfNotPresent @ vapi-background.js:1790
(anonymous) @ storage.js:140
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
µb.incrementRequestStats @ storage.js:148
journalProcess @ pagestore.js:743
(anonymous) @ pagestore.js:352
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
journalAddRequest @ pagestore.js:662
onBeforeRequest @ traffic.js:103
onBeforeSuspendableRequest @ vapi-background.js:1287
browser.webRequest.onBeforeRequest.addListener.denormalizeFilters.urls @ vapi-background.js:1239
subEventCallback @ extensions::webRequestEvent:106
webext.js:35 TypeError: Error in invocation of alarms.create(optional string name, alarms.AlarmCreateInfo alarmInfo): No matching signature.
    at webext.js:30:20
    at new Promise (<anonymous>)
    at Object.create (webext.js:28:16)
    at vapi-background.js:1792:27
(anonymous) @ webext.js:35
(anonymous) @ webext.js:28
(anonymous) @ vapi-background.js:1792
Promise.then (async)
createIfNotPresent @ vapi-background.js:1790
(anonymous) @ storage.js:140
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
µb.incrementRequestStats @ storage.js:148
journalProcess @ pagestore.js:743
(anonymous) @ pagestore.js:352
(anonymous) @ vapi-common.js:59
setTimeout (async)
on @ vapi-common.js:57
journalAddRequest @ pagestore.js:662
onBeforeRequest @ traffic.js:103
onBeforeSuspendableRequest @ vapi-background.js:1287
browser.webRequest.onBeforeRequest.addListener.denormalizeFilters.urls @ vapi-background.js:1239
subEventCallback @ extensions::webRequestEvent:106
```

**Comment by gorhill** [comment]
Despite those error messages, uBO appears to work fine though? The error messages are still shown by uBO itself, but they should no longer break uBO.

**Comment by Procyon-b** [comment]
I have never had any noticeable issue since switching to the beta since version 1.49.3.4 (it was my secondary computer at the time, but anyway...)
I only saw the errors since "Collect errors" is on by default for extensions locally installed.

**Comment by krystian3w** [comment]
> You seem to be correct, it's entirely BitWarden's fault.

Or in Firefox 116+, due on my ESR is not exist.

**Comment by valynor** [comment]
Brave Browser 1.65.133 Chromium: 124.0.6367.208 (Official Build) (arm64) on macOS 14.5 here. 
uBO 1.57.0

- same issue, suddenly basically no page loads anymore (browser renderer helper 100% cpu) 
- browser restart does not help, all pinned tabs do not load
- disabled all extensions .. browser worked again. This way I found out it's uBO causing this
- with uBO as the ONLY extension enabled .. nothing works. turning it off and immediately everything is fine again
- uninstalled, reinstalled uBO and set it up _exactly_ the same way as before ... everything still works

**Comment by xOleg01** [comment]
I have the same issue sometimes. uBO ext in Chromium Task Manager uses ~20% CPU and webpage is not loaded if its a first page after startup. uBO button freezing up too.
Browser restart and settings reload fixes the problem. No reinstall needed.

Chromium 122.0.6200.0 Iron Browser. Windows 10 LTSC. uBO 1.57.0

P.S. Chrome Web Store has a lot of complains on this problem (russian page).

**Comment by jimvlambe** [comment]
As of this morning (22nd May) the extension kills all web access to the internet. Switching the extension off restores function (Version 125.0.6422.76 (Official Build) (64-bit) Linux/Kubuntu).

**Comment by jscheid** [comment]
I'm seeing this issue on Firefox (macOS), I think it started yesterday. The symptoms are that the network seems to be down, i.e. everything takes forever to load. Once I turn off the extension it works. A browser restart (with the extension enabled) also helps, but only for some time. The last restart is a couple of hours ago perhaps and it started again just now.

<details><summary>Troubleshooting Information</summary>

```yaml
uBlock Origin: 1.58.0
Firefox: 127
filterset (summary):
 network: 138742
 cosmetic: 47005
 scriptlet: 20144
 html: 1851
listset (total-discarded, last-updated):
 default:
  user-filters: 0-0, never
  ublock-filters: 39099-112, 2h.14m Δ
  ublock-badware: 8705-0, 2h.14m Δ
  ublock-privacy: 1018-4, 2h.14m Δ
  ublock-unbreak: 2425-2, 2h.14m Δ
  ublock-quick-fixes: 119-5, 2h.14m Δ
  easylist: 86385-174, 2h.14m Δ
  easyprivacy: 52708-65, 2h.14m Δ
  urlhaus-1: 15240-5, 9m
  plowe-0: 3551-988, 5d.4h.56m
filterset (user): [empty]
trustedset:
 added: [array of 2 redacted]
userSettings: [none]
hiddenSettings: [none]
supportStats:
 allReadyAfter: 299 ms
 maxAssetCacheWait: 183 ms
 cacheBackend: indexedDB
```
</details>

**Comment by stephenhawk8054** [comment]
@jscheid Can you go to `about:addons` page (type in URL bar) and screenshot it here?

**Comment by jscheid** [comment]
@stephenhawk8054 I've React Developer Tools, Multi-Account Containers and 1Password. I haven't added or removed any add-ons recently, not for a month or so.

**Comment by stephenhawk8054** [comment]
@jscheid Can you turn off all other extensions, except uBO, restart the browser, open new tab and test again?

**Comment by jscheid** [comment]
The problem is it only happens spuriously. I could leave the others off for a day or so, but it would be an inconvenience and if the bug doesn't happen during that time it doesn't really prove much. (It may not have happened with the other addons enabled either.)

Counterproposal: I'm comfortable on the JS console, is there anything I can investigate next time it happens?

**Comment by gorhill** [comment]
You could launch a profiling session when the issue occurs to see if anything in the results is of use, see <https://profiler.firefox.com/>.

**Comment by stephenhawk8054** [comment]
If the issue occurs spuriously, how can you be sure it's fixed by turning off uBO but not another spurious event or by turning off other extensions?

If the issue occurs again, be sure that there's only uBO at that time, and use Firefox profiler as above link.

**Comment by stephenhawk8054** [comment]
> To me this strongly points at uBO as the offender especially given the other reports

This specific issue thread is specifically related to Chromium API, not related to Firefox.

**Comment by jscheid** [comment]
> This issue thread is specifically related to Chromium API, not related to Firefox.

Right, OK. I'll try to diagnose better next time it happens and happy to open a separate ticket then if I can still blame uBO 😄  Many thanks for your work on it by the way!

**Comment by garry-ut99** [comment]
Read https://github.com/uBlockOrigin/uBlock-issues/issues/3217#issuecomment-2106295005, your password manager extension or any other extension might be broken and slowing down uBO, disabling uBO, which fixes the issue, might mislead you thinking that it was uBO's fault, where it was not, if you read the thread, you would know it. If you don't want to disable other extensions, you can try to replace uBlock with [AdGuard](https://adguard.com/en/adguard-browser-extension/overview.html) for a while, to see whether the issue still occurs with other blocker or not, but it's not 100% guaranteed that the issue affects other blockers.

**Comment by Marocco2** [comment]
Ublock Origin 1.57.2 and 1.58.0 appears to be broken on latest Microsoft Edge Canary 128.0.2710.0
Tried to reinstall the extension but I can't get it to work and extension settings page is a white page

**Comment by gorhill** [comment]
Chrome/Edge Canary are not officially supported, too high likelihood of breakage. Use either beta or stable. ([example](https://github.com/AdguardTeam/AdguardBrowserExtension/issues/2869))

**Comment by rwong2888** [comment]
Edge Version 126.0.2592.102 (Official build) (arm64) stopped loading any page for me

**Comment by Villy-web** [comment]
> ```yaml
> githubusercontent.com
> ```
The problem seems to occur when the ublock fails to update its filter list. This problem consistently occurs for me when using a whitelist of sites on a proxy.

**Comment by Haus1** [comment]
I've had this happen two or three times in the past and the solution was to clear the cache and force it to resync. Thankfully it's been awhile since I've had it happen.

**Comment by garry-ut99** [comment]
It has happened for the first time for me as well, just right now:
- it has happened after previous uBO dev build has been uninstalled 
and then a new uBO dev build freshly installed (using "load unpacked extension")
- browser dev console - nothing, blank
- uBO background page - nothing, blank white page
- pages won't load - nothing, blank white page 
   - browser's circle icon spinning infinitely
   - in dev console Network pane: status "pending" 
 -  uBO toolbar icon:
    - clicking left mouse button - nothing, popup UI won't open
    - clicking right mouse button - one of menu's entries says "Failed to read or change website data"
 - while page is trying to load, Profiler freezes when trying to run it in console dev Performance pane, 
    -  until the page loading is manually stopped by "X" button, Profiler unfreezes then
    -  Profiler can be started before visiting page, but nothing is recored, as it's aborted instanlty after trying to visit a page, by "profiling overhead" event and most of spend time is described as "idle"

When in Debugger (dev console Source >> Workspace or Content Scripts]) I ticked both:
- "pause on uncaught exceptions"
- "pause on caught exceptions"

then clicked "Learn more" link, page was trying to load for a while, but then quickly some exception has occured:
`Paused on exception: TypeError: Failed to construct 'URL' : Invalid`: [SCREENSHOOT A](https://github.com/user-attachments/assets/b58eefda-66ab-4eb6-9421-a6ab45967b24) + [SCREENSHOOT B](https://github.com/user-attachments/assets/db387042-c9b3-4294-966f-280700d61343).

Then I noticed the same exception happens when trying to load any website.

**Comment by TsukiZero** [comment]
I would like to report EasyPrivacy as a frequent offender.

**Comment by DailyShana** [comment]
I encounter this problem again suddenly when using chrome, is there any tools to debug extensions?

**Comment by DailyShana** [comment]
the popup ui cannot be loaded either, if I remember correctly.

After posting comment here, I reinstall the ublock to resolve the problem "temporarily"

**Comment by gorhill** [comment]
You have other extensions? Other extensions can prevent uBO from initializing/responding in a timely manner. First step is to disable **all** other extensions and see if you can reproduce the issue.

**Comment by gorhill** [comment]
> 1password

All extensions run in the same process in Firefox, and if one extension hogs the CPU, all other extensions will no longer execute until the offending extension stops hogging the CPU. In the case of uBO, this means it has to wait before being able to tell the browser whether a network request should be filtered or not, and the symptom is pages/resources not loading as the browser waits for uBO to respond.

There was such an issue with 1Password recently, see <https://bugzilla.mozilla.org/show_bug.cgi?id=1980009>.
